### DISCUSSION TITLE
Guide for using context feature?

### DISCUSSION THREAD

**Discussion by sdondley** [kdheepak/taskwarrior-tui#428] [Q&A]
Impressive work.

I just got done adding a little shell magic so I can have a separate contexts in a separate shell process. TW doesn't do this out of the box.

 It would be cool if I could set that up with tui and have a separate tui terminal for each project I'm actively working on.

I tried to figure out how tui uses contexts but couldn't. When I add a new context and hit `c`, it doesn't show up in the popup window. Looked but didn't find any documentation.

**Comment by sdondley** [comment]
Oh, and I'm using TW 2.6.2

**Comment by sdondley** [comment]
What I tried was doing ! to get to shell and running `task context define test project:test` but nothing happens. I can run other tw commands from the command line like `td 1 done`

**Comment by sdondley** [comment]
OK, tui has to be shut down and re-opened before they appear.

**Reply by kdheepak** [reply]
You shouldn't need to shut down and re-open it though. Either press `r` to force refresh or wait for the next "tick" which is set at 250ms by default, and the TUI should reload all the contexts defined.
